### PR TITLE
feat: teams

### PR DESCRIPTION
**Pull request by prayash51**

### CODE REVIEW

File: storage/team/populate.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const team: Team = {
+      image: faker.image.imageUrl(),
+      name: {
+        first: faker.name.firstName(),
+        second: faker.name.middleName(),
+        last: faker.name.lastName(),
+      },
+      title: Tag[values[Math.floor(Math.random() * values.length)]],
+      description: faker.lorem.sentences(3),
+      tags: [],
```
Comment (OliwierOST) [inline storage/team/populate.ts:21]: add the id to the document, u can do that by calling
const teamsRef = collection(firestore, `teams`)
const teamRef = doc(teamsRef)
...
id: teamRef.id
...
setDoc(teamRef, team)

File: storage/team/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  image: URL
+  name: {
+    first: string
+    second: string
+    last: string
+  }
+  title: string
+  description: string
+  tags: Tag[]
+  _id?: UUID
```
Comment (OliwierOST) [inline storage/team/types.ts:24]: make the id required

**Review by OliwierOST — CHANGES_REQUESTED** [review]

File: components/Teams.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const { status, data: teams } = useFirestoreCollectionData(teamsQuery)
+
+  const handleClean = async () => {
+    const teamsDocs = await getDocs(collection(firestore, "teams"))
+    teamsDocs.forEach((doc) => deleteDoc(doc.ref))
+  }
+  return (
+    <>
+      <ButtonContainer>
+        <button onClick={() => populateTeams(firestore)}>
```
Comment (OliwierOST) [inline components/Teams.tsx:30]: We have some mock data already, so we need to drop the pattern of populating data on click

File: pages/index.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { Header } from "../components/Header"
 
 import { Footer } from "../components/Footer"
 import { Container, Box } from "@mui/material"
+import { Teams } from "components/Teams"
```
Comment (OliwierOST) [inline pages/index.tsx:5]: No absolute imports

File: storage/team/types.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  image: URL
+  name: {
+    first: string
+    second: string
+    last: string
+  }
+  title: string
+  description: string
+  tags: Tag[]
+  id: UUID
```
Comment (OliwierOST) [inline storage/team/types.ts:24]: nit: the `id` field should always be located at the top of the interface declaration, before any other fields

File: storage/team/types.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { UUID } from "./../common"
```
Comment (OliwierOST) [inline storage/team/types.ts:1]: Should be just `"../common"`

File: storage/team/populate.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    setDoc(teamRef, team)
+    alert("Team added")
+  } catch (e: any) {
```
Comment (OliwierOST) [inline storage/team/populate.ts:24]: Generally, `any` should not be used without justification

Please specify the types of errors one may expect

**Review by OliwierOST — CHANGES_REQUESTED** [review]

**Comment by OliwierOST** [comment]
One note for improving future workflow. Try to make commit messages more explanatory and don't be afraid to make multiple commits resolving the comments.

File: components/Teams.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+import Link from "next/link"
+import { useEffect } from "react"
+import styled from "styled-components"
+
+const ButtonContainer = styled.div`
+  display: flex;
+  gap: 4rem;
+  margin: 2rem 0rem;
+`
+const Image = styled.img`
```
[the remark this answers is not in the document]
Comment (prayash51) [reply components/Teams.tsx]: img tag changed to NextJS Image

File: storage/team/types.ts (hunk 5 above)
Comment (prayash51) [reply storage/team/types.ts]: Resolved

**Comment by OliwierOST** [comment]
Please have another look at CONTRIBUTING.md and try to spot what you can improve. And please stay consistent with other components and pages.

**Comment by piotrostr** [comment]
closing since it is outdated

**Pull request closed without merging**